src/sysusers/sysusers.c

```diff
@@ -533,6 +533,10 @@ static int write_temporary_shadow(const char *shadow_path, FILE **tmpfile, char
                                 ordered_hashmap_remove(todo_uids, UID_TO_PTR(i->uid));
                         }
 
+                        /* Make sure we keep the NIS entries (if any) at the end. */
+                        if (IN_SET(sp->sp_namp[0], '+', '-'))
+                                break;
+
                         errno = 0;
```

Comment:
ok, i am officially an idiot. I just asked you to move the errno down, but actually your original code was right. Because we check errno right after this loop, and in that case we should not trip up on any side-effect of fgetspent()...

So sorry for the forth and back!

Please move the errno initializuation before the if check again!